Modal para el detalle del pedido

rafael1199v opened the pull request:
1. Se descargaron los cambios del ultimo commit y se unió a la funcionalidad construida.
2. Se creó un componente para el detalle del pedido en base al diseño realizado.
3. Se crearon iconos como componentes para separa y modularizar mas el codigo.
4. Se realizó la conexion con la base de datos de supabase y el archivo base para la estructura de las variables de entorno.
5. Se eliminaron los contenidos del index.css y App.css para usar solo los estilos de tailwind.

Se piensa mejorar la estructura del modal en futuros commits para poder ser reutilizado en otras ocasiones. Como tambien agregar una función para conectarte a la base de datos.

Flujo principal:
1. Lista de pedidos
![image](https://github.com/user-attachments/assets/549fd529-1580-4715-95ca-db72ba8ce5a7)
2. Abrir el detalle de un pedido
![image](https://github.com/user-attachments/assets/77a02f86-88c8-4b7a-87bc-09cd015c18f4)

rafael1199v committed:
Modal para el detalle del pedido

1. Se descargaron los cambios del ultimo commit y se unió a la funcionalidad construida.
2. Se creó un componente para el detalle del pedido en base al diseño realizado.
3. Se crearon iconos como componentes para separa y modularizar mas el codigo.
4. Se realizó la conexion con la base de datos de supabase y el archivo base para la estructura de las variables de entorno.
5. Se eliminaron los contenidos del index.css y App.css para usar solo los estilos de tailwind.

Se piensa mejorar la estructura del modal en futuros commits para poder ser reutilizado en otras ocasiones.
Como tambien agregar una función para conectarte a la base de datos

rafael1199v commented:
Visualización en teléfono:
![image](https://github.com/user-attachments/assets/043268fe-71b4-49d0-85ca-b9877fd90efe)

Cristopher8049 commented on OrderNow-Dashboard/src/components/order-detail/OrderDetail.jsx:
```diff
@@ -0,0 +1,122 @@
+const supabaseKey = import.meta.env.VITE_SUPABASE_ANON_KEY;
+
+function OrderDetail({orderId, closeModal}) {
+  const [orderDetail, setOrderDetail] = useState({});
```
Es recomendable crear el cliente de Supabase fuera del componente, para evitar que se cree nuevamente cada vez que el componente se renderiza. Esto se puede hacer moviendo la creación del cliente fuera del componente OrderDetail. Ejemplo:
OrderNow-Dashboard/src/config/supabaseClient.js

Cristopher8049 commented on OrderNow-Dashboard/src/components/order-detail/OrderDetail.jsx:
```diff
@@ -0,0 +1,122 @@
+          order_details(
+            id,
+            products(
+              name,
```
El componente no tiene ningún tipo de manejo de estados de carga o error para la llamada a la API. Sería bueno mostrar un indicador de carga mientras se obtiene la información y un mensaje de error en caso de que falle la solicitud.

Cristopher8049 commented on OrderNow-Dashboard/src/components/order-detail/OrderDetail.jsx:
```diff
@@ -0,0 +1,122 @@
+    fetchBusiness();
+  }, []);
+
+  const fetchBusiness = async () => {
+    const { data, error } = await supaBaseCom
+      .from("orders")
+      .select(
+        `
+          total_price,
+          order_details(
+            id,
+            products(
+              name,
+              price
+            ),
+            quantity
+          )
+        `
+      )
+      .eq("id", orderId)
+      .single();
+
+    if (error) {
+      alert(error.message);
+      return;
+    }
+    
+    setOrderDetail(data);
+  };
```
El componente no tiene ningún tipo de manejo de estados de carga o error para la llamada a la API. Sería bueno mostrar un indicador de carga mientras se obtiene la información y un mensaje de error en caso de que falle la solicitud.

rafael1199v replied:
![image](https://github.com/user-attachments/assets/a100b4d0-467c-434f-90e4-e87e1efe30ab)
Resuelto

Cristopher8049 commented on OrderNow-Dashboard/src/components/order-detail/OrderDetail.jsx:
```diff
@@ -0,0 +1,122 @@
+import { createClient } from "@supabase/supabase-js";
+import CloseIcon from "../../subcomponents/icons/CloseIcon"
+
+const supabaseUrl = import.meta.env.VITE_SUPABASE_URL;
+const supabaseKey = import.meta.env.VITE_SUPABASE_ANON_KEY;
+
+function OrderDetail({orderId, closeModal}) {
+  const [orderDetail, setOrderDetail] = useState({});
+
+  const supaBaseCom = createClient(supabaseUrl, supabaseKey, {
+    db: { schema: "com" },
+  });
```
Es recomendable crear el cliente de Supabase fuera del componente, para evitar que se cree nuevamente cada vez que el componente se renderiza. Esto se puede hacer moviendo la creación del cliente fuera del componente OrderDetail. Ejemplo:
OrderNow-Dashboard/src/config/supabaseClient.js

rafael1199v replied:
Se creó una carpeta y el archivo para obtener el cliente de supabase.

rafael1199v committed:
Refactorización: Conexión con supabase y mejora de UX

1. Se añadió una carpeta para la obtención del cliente de supabase para separar responsabilidades.
2. Se añadieron estados de carga y error para tener una mejor experiencia de usuario

rafael1199v committed:
Merge remote-tracking branch 'upstream/main'

rafael1199v committed:
Hotfix: Libreria react-router-dom

Se instaló la libreria de react-router-dom ya que no aparecía en el archivo package.json y es usado para el enrutamiento

rafael1199v commented:
Conflictos resueltos @RaidenMT 
![image](https://github.com/user-attachments/assets/080c4582-5578-4dbb-becc-b53332b4c4a9)
